systemtest/src/test/java/io/strimzi/systemtest/specific/SpecificIsolatedST.java

```diff
@@ -193,6 +207,97 @@ void testLoadBalancerSourceRanges(ExtensionContext extensionContext) {
+            ClusterRoleBindingTemplates.getClusterOperatorWatchedCrb(clusterOperator.getClusterOperatorName(), clusterOperator.getDeploymentNamespace()),
+            ClusterRoleBindingTemplates.getClusterOperatorEntityOperatorCrb(clusterOperator.getClusterOperatorName(), clusterOperator.getDeploymentNamespace())
+        );
+
+        clusterOperator.unInstall();
+        // create namespace, where we will be able to deploy Custom Resources
+        cluster.createNamespace(CollectorElement.createCollectorElement(extensionContext.getRequiredTestClass().getName(), extensionContext.getRequiredTestMethod().getName()), namespaceWhereCreationOfCustomResourcesIsApproved);
+        clusterOperator = clusterOperator.defaultInstallation()
+            .withWatchingNamespaces(Constants.WATCH_ALL_NAMESPACES)
+            // use our pre-defined Roles
+            .withRoles(roles)
+            // use our pre-defined RoleBindings
+            .withRoleBindings(roleBindings)
+            // use our pre-defined ClusterRoles
+            .withClusterRoles(clusterRoles)
+            // use our pre-defined ClusterRoleBindings
+            .withClusterRoleBindings(clusterRoleBindings)
+            .createInstallation()
+            .runBundleInstallation();
+
+        // implicit verification that a user is able to deploy Kafka cluster in namespace <example-1>, where we are allowed
+        // to create Custom Resources because of `*-namespaced Role`
+        resourceManager.createResource(extensionContext, false, KafkaTemplates.kafkaEphemeral(testStorage.getClusterName(), 3)
+            .editMetadata()
+            // this should work
+            .withNamespace(namespaceWhereCreationOfCustomResourcesIsApproved)
+            .endMetadata()
+            .build());
+
+        resourceManager.createResource(extensionContext, false, KafkaTemplates.kafkaEphemeral(testStorage.getClusterName(), 3)
+            .editMetadata()
+                // this should not work
+                .withNamespace(clusterOperator.getDeploymentNamespace())
+            .endMetadata()
+            .build());
+
+        // verify that in `infra-namespace` we are not able to deploy Kafka cluster
+        KafkaUtils.waitForKafkaStatusUpdate(clusterOperator.getDeploymentNamespace(), testStorage.getClusterName());
+
+        final Condition condition = KafkaResource.kafkaClient().inNamespace(clusterOperator.getDeploymentNamespace()).withName(testStorage.getClusterName()).get().getStatus().getConditions().stream().findFirst().get();
+
+        assertThat(condition.getReason(), CoreMatchers.is("KubernetesClientException"));
+        assertThat(condition.getStatus(), CoreMatchers.is("True"));
+        assertThat(condition.getMessage(), CoreMatchers.containsString("Forbidden!Configured service account doesn't have access"));
+
+        // rollback to the default configuration
+        clusterOperator.rollbackToDefaultConfiguration();
```

Comment:
What exactly does this mean?

Reply:
Well, this actually un-install the Cluster Operator with the specific configuration, which we used in the test case, and then deploys the default Cluster Operator.

Reply:
Maybe the comment can be more explicit about it then? How does it know what to uninstall if you install the RBAC bindings in your own way? How do you make sure they are not left behind?

Reply:
> Maybe the comment can be more explicit about it then? 

Yes, I will more elaborate on this comment :) thanks

> How does it know what to uninstall if you install the RBAC bindings in your own way? How do you make sure they are not left behind?

TL;DR; It is handled by ResourceManager inside SetupClusterOperator.
--
When you create a resource by ResourceManager using the generic method `resourceManager.createResource(extensionContent, <resource>)` it will do the following:
1. Find resource type; meaning if such resource is Kafka, KMM2 etc.
2. Create a resource 
3. and then we store info to the `Map<String, Stack<ResourceItem>>`, the key identifier is the test case name and the Stack contains multiple resources, which will be deleted after the test case because we call `ResourceManager.getInstance().deleteResources(extensionContext);` method, which retrieve all of the resources specific to test case thanks to extensionContent metadata.

In this scenario `clusterOperator.rollbackToDefaultConfiguration();` calls at first unInstall() method, which completely deletes all resources (this is handled by ResourceManager) and all namespaces. Thus also removing RBAC bindings deployed in this test case.

Reply:
Ok, thanks for the explanation. 👍